Library/OcCpuLib/OcCpuLib.c

```diff
@@ -1252,8 +1260,12 @@ ScanAmdProcessor (
     //
     Cpu->CurBusRatio = Cpu->MaxBusRatio;
     Cpu->MinBusRatio = Cpu->MaxBusRatio;
-
-    Cpu->FSBFrequency = DivU64x32 (Cpu->CPUFrequency, Cpu->MaxBusRatio);
+    if (Cpu->MaxBusRatio == 0 || Cpu->Hypervisor) {
+        // if hypervisor is true, then we just set FSBFrequency
```

Comment:
This branch is misleading a lot. The whole reason, as I understood it, of the branches above, is to support 0 values from HVs, so for our current knowledge, those two conditions should describe the same state, for which one way *has* to be chosen for a proper workaround design.
Either you allow *both* bare metal and HVs to report 0 values and use it as sole condition, ignoring the virtualisation aspect, or you allow 0 values only from HVs and use virtualisation as the condition. Latter has the disadvantage that when a HV actually provides sane values, they are discarded, as it is the case with your code here too.

Reply:
This branch should most likely be rewritten as follows:

```
if (Cpu->MaxBusRatio == 0) {
   ASSERT (Cpu->Hypervisor); ///< Currently the only known case with 0 MaxBusRatio is hypervisor
   ...
}
```